src/CMakeLists.txt

```diff
@@ -328,7 +328,11 @@ foreach(_module ${HPX_LIBS})
   if(UNIX OR APPLE)
     # add the full archive (some are not used directly in libhpx and need to be
     # forced into libhpx with this option)
+  if(APPLE)
+    set(_module_target "-Wl,-all_load" "${_module_target}")
+  else(APPLE)
```

Comment:
Sorry for the nitpick. Could we do:
```
if(UNIX)
  set(... whole-archive ...)
elseif(APPLE)
  set(... all_load ...)
endif()
```
instead and skip the nested ifs?

Reply:
@msimberg done